src/ActionParameterHandler.ts

```diff
@@ -103,43 +105,83 @@ export class ActionParameterHandler<T extends BaseDriver> {
     /**
      * Normalizes parameter value.
      */
-    protected normalizeParamValue(value: any, param: ParamMetadata): Promise<any>|any {
+    protected async normalizeParamValue(value: any, param: ParamMetadata): Promise<any> {
         if (value === null || value === undefined)
             return value;
 
-        switch (param.targetName) {
+        // if param value is an object and param type match, normalize its string properties
+        if (typeof value === "object" && ["queries", "headers", "path-params", "cookies"].indexOf(param.type) !== -1) {
```

Comment:
`.some` is easier to read than `.indexOf` so please `["queries", "headers", "path-params", "cookies"].indexOf(param.type) !== -1` => `["queries", "headers", "path-params", "cookies"].some(type => type == param.type)`